### ISSUE TITLE
OrderBy Sort on more Type?

### ISSUE THREAD

**Issue by BallisticPain** [danrevah/ngx-pipes#25]
Thanks for adding the OrderBy, however, I'm still unable to swap to using ngx-pipes as your OrderBy only handles strings essentially. It needs to be able to handle more data types. Dates, Booleans, Numbers, etc... I added sorting booleans to the one I linked in the previous ticket.

```
    if(isBoolean(a) || isBoolean(b)) {
      if (a == b) return 0;
      if (a && !b) return -1;
      if (!a && b) return 1;
    }
```

Thanks!
Jarvis

**Comment by danrevah** [comment]
It does support date, booleans, numbers and strings.
Can you show me exactly what's not working?

Take a look at the tests I've added (they work with dates & boolean values, and there's older tests for numbers):
https://github.com/danrevah/ngx-pipes/commit/1e709bdea2c77fcb8aa0f52f0c27b7d128c1f059

**Comment by danrevah** [comment]
Since you haven't responded yet I have been digging into this, trying to find what isn't working properly. I found some issue and fixed it, it might be related to your issue as well.
Its released to version 1.4.3.

Update me to let me know if that fix your case.
If not, please provide more details and an example of your object and usage.

**Comment by danrevah** [comment]
It's been 10 days and you haven't responded.
I'm closing this issue for now.

Thanks.
